Active Record Callbacks > 6. Halting Execution (#379)
campuscode/rails-guides-pt-BR: pt-BR/active_record_callbacks.md

These methods should be used with caution, however, because important business rules and application logic may be kept in callbacks. Bypassing them without understanding the potential implications may lead to invalid data.

> Terminando uma Execução

Reviewer: ```suggestion
Interrompendo uma Execução
```
Acho que aqui está mais para o contexto de "interromper" do que "terminar". "Terminar" dá um sentido de "finalizar". O que acha?

Author: Sim, me faltava essa palavra haha. Muito obrigado @tkusuki :) 👍

-----------------

> Quando você começar a registrar novos *callbacks* para seus modelos, eles serão enfileirados para a execução. Esta fila incluirá todas as validações do seu modelo, os *callbacks* registrados, e a operação do banco de dados para ser executada.

Reviewer: ```suggestion
Quando você começa a registrar novos *callbacks* para seus *models*, eles serão enfileirados para a execução. Esta fila incluirá todas as validações do seu *model*, os *callbacks* registrados e a operação do banco de dados a ser executada.
```

> Toda a cadeia do *callback* é empacotada numa transação. Se algum *callback* lança uma exceção, a cadeia de execução é terminada e um *ROLLBACK* é emitido. Para parar intencionalmente uma cadeia, use:

Reviewer: ```suggestion
Toda a cadeia do *callback* é empacotada em uma transação. Se algum *callback* lança uma exceção, a cadeia de execução é interrompida e um *ROLLBACK* é emitido. Para interromper intencionalmente uma cadeia, use:
```

```ruby
throw :abort
```

> WARNING: Qualquer exceção que não as `ActiveRecord::Rollback` ou `ActiveRecord::RecordInvalid` serão re-lançadas pelo Rails após a cadeia do *callback* ser terminada. Lançar uma outra exceção que não `ActiveRecord::Rollback` ou `ActiveRecord::RecordInvalid` pode quebrar um código que não espera por métodos como `save` ou `update` (os quais normalmente tentam retornar `true` ou `false`) para lançar uma exceção.

Reviewer: ```suggestion
WARNING: Qualquer exceção que não seja `ActiveRecord::Rollback` ou `ActiveRecord::RecordInvalid` serão lançadas novamente pelo Rails após a cadeia do *callback* ser interrompida. Lançar uma outra exceção que não `ActiveRecord::Rollback` ou `ActiveRecord::RecordInvalid` pode quebrar um código que não espera por métodos como `save` ou `update` (os quais normalmente tentam retornar `true` ou `false`) para lançar uma exceção.
```
